yaqd_andor/_andorsdk2_ixon.py

```diff
@@ -0,0 +1,309 @@
+            self.spec_order = None
+            self.spec_focal_length = None
+            self.spec_calibration_pixel = None
+
+    def _gen_mappings(self):
+        """Generate map."""
+
+        if self.has_mono:
+            # translate inputs into appropriate internal units
+            spec_inclusion_angle_rad = 0.00
+            # spec_inclusion_angle_rad = np.radians(float(self._config["inclusion_angle"]))
+            spec_focal_length_tilt_rad = 0.00
+            # spec_focal_length_tilt_rad = np.radians(float(self._config["focal_length_tilt"]))
+            pixel_width_mm = float(self.sensor_width / self.max_width) / 1000.00
+
+            self._channel_mappings = {"image": ["wavelengths", "y_index"]}
+            self._mapping_units = {"wavelengths": "nm", "y_index": "None"}
+
+            # create array
+            i_pixel = np.array(self.x_ai, dtype=float)
+            eff_pixel_width_mm = float(int(self.binning) * pixel_width_mm)
+            # calculate terms
+            x = np.arcsin(
+                (1e-6 * self.spec_order * self.spec_grooves_per_mm * self.spec_position)
+                / (2 * np.cos(spec_inclusion_angle_rad / 2.0))
+            )
+            A = np.sin(x - spec_inclusion_angle_rad / 2)
+            B = np.sin(
+                (spec_inclusion_angle_rad)
+                + x
+                - (spec_inclusion_angle_rad / 2)
+                - np.arctan(
+                    (
+                        eff_pixel_width_mm * (i_pixel - self.spec_calibration_pixel)
+                        + self.spec_focal_length * spec_focal_length_tilt_rad
+                    )
+                    / (self.spec_focal_length * np.cos(spec_focal_length_tilt_rad))
+                )
+            )
+            out = ((A + B) * float(1e6)) / (
+                float(self.spec_order) * float(self.spec_grooves_per_mm)
+            )
+            self._mappings = {"x_index": out, "y_index": self.y_ai}
+        else:
+            self._channel_mappings = {"image": ["x_index", "y_index"]}
+            self._mapping_units = {"x_index": "None", "y_index": "None"}
+            self._mappings = {"x_index": self.x_ai, "y_index": self.y_ai}
+        return
+
+    def _set_aoi(self):
+        aoi_keys = ["aoi_binning", "aoi_width", "aoi_left", "aoi_height", "aoi_top"]
+        binning, width, left, height, top = [self._config[k] for k in aoi_keys]
+        left = int(left)
+        width = int(width)
+        height = int(height)
+        top = int(top)
+        binning = int(binning)  # equal xy binning, so only need 1 index
+
+        # handle defaults (maximum sizes)
+        if left is None:
+            left = 1
+        if top is None:
+            top = 1
+        if width is None:
+            width = self.max_width - left + 1
+        if height is None:
+            height = self.max_height - top + 1
+
+        arrwidth = int(int(width) / int(binning))
+        arrheight = int(int(height) / int(binning))
```

Comment:
```suggestion
        arrwidth = width // binning
        arrheight = height // binning
```
you already typed width, height, binning as ints (above), so no need to do again

Reply:
ok